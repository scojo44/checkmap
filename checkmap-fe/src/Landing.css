.Landing  {
  /* position: absolute;
  inset: 0;
  z-index: 9999;
  background-color: rgba(0, 0, 0, 50%); */
}
.Landing .welcome {
  /* position: relative;
  top: 10%;
  width: 50%;
  min-width: '400px';
  margin: auto;
  background-color: rgba(255, 255, 255, 70%);
  border-radius: 1rem;
  padding: 1rem; */
}
.Landing h1 {
  font-family: Lora;
  text-align: center;
  font-size: 3rem;
  margin-top: 0;
}
.Landing form {
  margin: 0;
}
.Landing a {
  text-decoration: none;
}
.Landing a:link,
.Landing a:visited {
  color: unset;
}