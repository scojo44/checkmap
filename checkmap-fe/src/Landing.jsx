import React from 'react'
import {Link} from 'react-router-dom';
import './Landing.css'

export default function Landing() {
  return (
    <section className="Landing">
      <h1>CheckMap</h1>
      <h2>Welcome!</h2>
      <p>CheckMap is a simple site to track states and counties you've visited.</p>
      <p>Sign up, create a list, then just click on the regions you have visited!</p>
      <p>You can create more lists to track where you have done something special like
        caught a fish, played golf, rode your motorcycle, found a geocache...</p>
      <form>
        <button><Link to="/signup">Sign Up</Link></button>
        <button><Link to="/login">Log In</Link></button>
      </form>
    </section>
  );
}
