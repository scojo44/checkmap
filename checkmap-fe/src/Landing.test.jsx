import {render} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import UserContext from './UserContext'
import Landing from './Landing'

const context = {
  user: false // Only used to see if user logged in and should navigate to users area
};

describe('Landing Tests', () => {
  it('Renders without crashing', () => {
    render(
      <MemoryRouter>
        <UserContext.Provider value={context}>
          <Landing />
        </UserContext.Provider>
      </MemoryRouter>
    );
  });

  it('Matches snapshot', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <UserContext.Provider value={context}>
          <Landing />
        </UserContext.Provider>
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
