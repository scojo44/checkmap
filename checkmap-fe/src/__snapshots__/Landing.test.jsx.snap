// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Landing Tests > Matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="Landing"
  >
    <h1>
      CheckMap
    </h1>
    <h2>
      Welcome!
    </h2>
    <p>
      CheckMap is a simple site to track states and counties you've visited.
    </p>
    <p>
      Sign up, create a list, then just click on the regions you have visited!
    </p>
    <p>
      You can create more lists to track where you have done something special like caught a fish, played golf, rode your motorcycle, found a geocache...
    </p>
    <form>
      <button>
        <a
          href="/signup"
        >
          Sign Up
        </a>
      </button>
      <button>
        <a
          href="/login"
        >
          Log In
        </a>
      </button>
    </form>
  </section>
</DocumentFragment>
`;
